lib/temporal.rb

```diff
@@ -158,6 +216,11 @@ def metrics
       @metrics ||= Metrics.new(configuration.metrics_adapter)
     end
 
+    class ResultConverter
```

Comment:
For those curious, I used a combination of responses from here: https://stackoverflow.com/questions/39716799/how-to-implement-private-inner-class-in-ruby